### PR TITLE
Add state logging

### PR DESCRIPTION
**Pull request by spdskatr**

### CODE REVIEW

File: gym_sts/data/state_log_loader.py [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+from typing import TextIO
+
+import numpy as np
+from tqdm import tqdm
+
+from gym_sts.spaces.actions import ACTIONS
+from gym_sts.spaces.observations import Observation
+
+
+# TODO: Remove this after everything in the main obs space outputs numpy arrays
```
Comment (vladfi1) [inline gym_sts/data/state_log_loader.py]: You can remove this now.

File: gym_sts/envs/base.py [hunk 2]
```diff
@@ -470,6 +482,10 @@ def valid_actions(self) -> list[Action]:
             raise RuntimeError("Game not started?")
         return latest_obs.valid_actions
 
+    def save_artifact(self, file_name):
+        # TODO: Implement uploading to WandB
+        raise Exception("Not implemented")
```
Comment (vladfi1) [inline gym_sts/envs/base.py:487]: Exception -> NotImplementedError

**Review by kronion — APPROVED** [review]

File: gym_sts/data/state_log_loader.py (hunk 1 above)
Comment (spdskatr) [reply gym_sts/data/state_log_loader.py]: Done

File: gym_sts/envs/base.py (hunk 2 above)
Comment (spdskatr) [reply gym_sts/envs/base.py:487]: Done

**Pull request merged**